src/Algolia.Search.Test/EndToEnd/Index/SettingsTest.cs

```diff
@@ -184,6 +183,15 @@ public async Task TestSettings()
             Assert.True(TestHelper.AreObjectsEqual(settings, getSettingsResponseAfterChanges,
                 spceficPropertiesCheck.ToArray()));
 
+            // Set new value for NumericAttributesForFiltering
+            settings.NumericAttributesForFiltering = null;
+
+            var saveSettingsResponseAfterChangesNumericAttributesForFiltering = await _index.SetSettingsAsync(settings);
+            saveSettingsResponseAfterChangesNumericAttributesForFiltering.Wait();
+
+            var getSettingsResponseAfterChangesNumericAttributesForFiltering = await _index.GetSettingsAsync();
+            Assert.AreEqual(null, getSettingsResponseAfterChangesNumericAttributesForFiltering.NumericAttributesForFiltering);
+
```

Comment:
Please add another line when you are setting `NumericAttributesForFiltering` to a `List<String>`, like that we have the full loop. `string[]` => `null` => `string[]`.